### PR TITLE
feat(TIFFReader): add vtkTIFFReader

### LINKED ISSUE #3169: [Feature] Support for TIFF images
### Motivation

Support for [Tagged Image File Format](https://en.wikipedia.org/wiki/Tagged_Image_File_Format) (TIFF) to load textures.

### Detailed Description

Add a new vtkTIFFReader

```js
const reader = vtkTIFFReader.newInstance();
reader.parse(fileReader.result);
const imageData = reader.getOutputData();
```

### PR DESCRIPTION
**Pull request by daker**
This PR adds support for loading TIFF image format

### Context
The reader is using [UTIF.js](https://github.com/photopea/UTIF.js) from Photopea

fixes #3169

### Results

https://github.com/user-attachments/assets/544cc382-bf9a-49e9-862d-77a6ba710eed

### Changes
- [x] Documentation and TypeScript definitions were updated to match those changes

### PR and Code Checklist
- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Testing
- [x] This change adds or fixes unit tests 
- [x] Tested environment:
  - **vtk.js**: 
  - **OS**: 
  - **Browser**:

### CODE REVIEW

File: Sources/IO/Image/TIFFReader/index.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+  macro.get(publicAPI, model, ['url', 'baseURL']);
+  macro.setGet(publicAPI, model, ['dataAccessHelper', 'flipY']);
+
+  // Object specific methods
+  vtkTIFFReader(publicAPI, model);
+
+  // To support destructuring
+  if (!model.compression) {
+    model.compression = null;
```
Comment (finetjul) [inline Sources/IO/Image/TIFFReader/index.js:142]: not sure why you need to enforce setting it to null.
Why undefined is not ok?
One way to do it is to specify compression in DEFAULT_VALUES?

Comment (daker) [reply Sources/IO/Image/TIFFReader/index.js:142]: that's how it's done in other readers, do you want me to change it ? 

https://github.com/Kitware/vtk-js/blob/master/Sources/IO/Geometry/STLReader/index.js#L360-L367
https://github.com/Kitware/vtk-js/blob/master/Sources/IO/Geometry/DracoReader/index.js#L323-L330

Comment (finetjul) [reply Sources/IO/Image/TIFFReader/index.js:142]: it seems that it is to allow `const { compression, progressCallback } = model;` in the DracoReader.

I suggest you move them all in DEFAULT_VALUE.

**Comment by finetjul** [comment]
Can you please add a screenshot in the example gallery ?

File: package-lock.json [hunk 2]
```diff
@@ -16519,6 +16520,11 @@
         "node": ">=6"
       }
     },
+    "node_modules/pako": {
+      "version": "1.0.11",
+      "resolved": "https://registry.npmjs.org/pako/-/pako-1.0.11.tgz",
+      "integrity": "sha512-4hLB8Py4zZce5s4yd9XzopqwVv/yGNhV1Bl8NTmCq1763HeK2+EwVTv+leGeL13Dnh2wfbqowVPXCIO0z4taYw=="
+    },
```
Comment (finetjul) [inline package-lock.json:16527]: is this really needed?

Comment (daker) [reply package-lock.json:16527]: it's used by UTIF 
https://github.com/photopea/UTIF.js/blob/master/package.json#L22
https://github.com/photopea/UTIF.js/blob/master/UTIF.js#L44

Comment (finetjul) [reply package-lock.json:16527]: gotcha

**Review by finetjul — APPROVED** [review]
LGTM (if we are ok with the new utif dependency). @floryst ?

**Comment by floryst** [comment]
I'm fine with the `utif` dependency. Maybe better to mark it as an optionalDependency, but that won't change much of anything, except for ignoring errors if `utif` can't be resolved.

**Pull request merged**